lib/mongoid/relations/counter_cache.rb

```diff
@@ -103,6 +103,24 @@ def add_counter_cache_callbacks(meta)
           name = meta.name
           cache_column = meta.counter_cache_column_name.to_sym
 
+          after_update do
+            if record = __send__(name)
+
+              if __send__("#{name}_id_changed?".to_sym)
+                original, current = __send__("#{name}_id_change".to_sym)
```

Comment:
@durran are there other methods I can use instead of
__send__("#{name}_id_changed?".to_sym) 
and
__send__("#{name}_id_change".to_sym)